cmd/check_pow.cpp

```diff
@@ -131,15 +131,16 @@ int main(int argc, char* argv[]) {
 
             auto boundary256{header->boundary()};
             auto seal_hash(header->hash(/*for_sealing =*/true));
-            ethash::hash256 sealh256{*reinterpret_cast<ethash::hash256*>(seal_hash.bytes)};
-            ethash::hash256 mixh256{*reinterpret_cast<ethash::hash256*>(header->mix_hash.bytes)};
-            if (!ethash::verify(*epoch_context, sealh256, mixh256, nonce, boundary256)) {
+            const auto sealh256{ethash::hash256_from_bytes(seal_hash.bytes)};
+            const auto mixh256{ethash::hash256_from_bytes(header->mix_hash.bytes)};
+            if (const auto ec = ethash::verify(*epoch_context, sealh256, mixh256, nonce, boundary256); ec) {
                 auto result{ethash::hash(*epoch_context, sealh256, nonce)};
                 auto b{to_bytes32({boundary256.bytes, 32})};
                 auto f{to_bytes32({result.final_hash.bytes, 32})};
                 auto m{to_bytes32({result.mix_hash.bytes, 32})};
 
                 std::cout << "\n Pow Verification error on block " << block_num << " : \n"
+                          << "Error: " << ec << "\n"
```

Comment:
I'm not sure where this is used. But here you can either print final or mix hash, depending on error code.

Reply:
Not a big deal. `check_pow` is only a debug tool